ledger/participant-integration-api/src/main/scala/platform/store/backend/common/MeteringStorageBackendTemplate.scala

```diff
@@ -66,3 +77,113 @@ private[backend] class MeteringStorageBackendTemplate(ledgerEndCache: LedgerEndC
   private def isSet(o: Option[_]): Int = o.fold(0)(_ => 1)
 
 }
+private[backend] class MeteringStorageBackendWriteTemplate() extends MeteringStorageWriteBackend {
+
+  val participantMeteringParser: RowParser[ParticipantMetering] = {
+    (
+      applicationId("application_id") ~
+        timestampFromMicros("from_timestamp") ~
+        timestampFromMicros("to_timestamp") ~
+        int("action_count") ~
+        offset("ledger_offset")
+    ).map {
+      case applicationId ~
+          from ~
+          to ~
+          actionCount ~
+          ledgerOffset =>
+        ParticipantMetering(
+          applicationId,
+          from,
+          to,
+          actionCount,
+          ledgerOffset,
+        )
+    }
+
+  }
+
+  def transactionMeteringMinTimestamp(
+      connection: Connection
+  ): Option[Timestamp] = {
+    SQL"""select min(metering_timestamp) from transaction_metering"""
+      .as(timestampFromMicros(1).?.single)(connection)
+  }
+
+  def transactionMeteringMaxOffset(from: Offset, to: Timestamp)(
+      connection: Connection
+  ): Option[Offset] = {
+
+    import com.daml.platform.store.Conversions.OffsetToStatement
+    import com.daml.platform.store.Conversions.TimestampToStatement
+
+    SQL"""
+      select max(ledger_offset)
+      from transaction_metering
+      where ledger_offset > $from
+      and metering_timestamp < $to
```

Comment:
I'm expecting a table scan as:
-  ledger_offset > $from will do nothing as the from will be earlier than any table rows
- metering_timestamp < $to will include most of the rows in the table.
This will only be scanning one hour of data.